zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSetting.java

```diff
@@ -281,9 +275,14 @@ public InterpreterSetting(InterpreterSetting o) {
     this.conf = o.getConf();
   }
 
-  private void createLauncher() {
+  private void createLauncher(Properties properties) {
     if (group.equals("spark")) {
-      this.launcher = new SparkInterpreterLauncher(this.conf);
+      if (properties.getProperty("master") != null
+              && properties.getProperty("master").startsWith("k8s")) {
+        this.launcher = new SparkK8InterpreterLauncher(this.conf);
+      } else {
+        this.launcher = new SparkInterpreterLauncher(this.conf);
```

Comment:
@zjffdu - I think we need a fast way to abstract out the different launchers, instead of bringing everything in here (plus artifacts like kubernetes-client). Do you have any thought on that?

Reply:
Eventually I was thinking about this as a long term solution - applicable not just to this launcher. If there was something like a pluggable/configurable launcher, we don't have to include the k8s client and SparkK8Launcher in core zeppelin code,  just implement as a plugabble launcher added to Zeppelin classpath and configure Zeppelin to use it.